xdsl/pattern_rewriter.py

```diff
@@ -126,6 +126,8 @@ def _can_modify_region(self, region: Region) -> bool:
         """Check if the region and its children can be modified by this rewriter."""
         if region.parent is None:
             return True  # Toplevel operation of current_operation is always a ModuleOp
+        if region is self.current_operation.parent_region():
+            return True
```

Comment:
@math-fehr, to make the test pass, I had to add this line. I feel that it makes sense, we should be able to modify the regions that are the direct parents of the matched operation.